### PR TITLE
Cleanup HermiteInitialGuesses

### PR DESCRIPTION
**Pull request by btracey**

### CODE REVIEW

**Comment by btracey** [comment]
Initial set of cleanups before the major surgery on this function in the next commit.

**Comment by btracey** [comment]
PTAL @vladimir-ch

File: quad/hermite.go [hunk 1]
```diff
@@ -127,8 +126,9 @@ func (h Hermite) locationsAsy(n int) (x, w []float64) {
 
 func (h Hermite) locationsAsy0(n int) (x, w []float64) {
 	eps := math.Nextafter(1, math.Inf(1)) - 1
+	x0pts := make([]float64, n/2+n%2)
```
Comment (vladimir-ch) [inline quad/hermite.go]: Would it be worth making it a field in Hermite for reuse?

File: quad/hermite.go [hunk 2]
```diff
@@ -253,80 +253,75 @@ func (h Hermite) hermpolyAsyAiry(n int, theta []float64) (valVec, dvalVec []floa
...
 			0.2*math.Pow(2, 4.0/3)*ar*ar*math.Pow(nu, -1.0/3) +
 			(11.0/35-a*a-12.0/175*ar*ar*ar)/nu +
 			(16.0/1575*ar+92.0/7875*math.Pow(ar, 4))*math.Pow(2, 2.0/3)*math.Pow(nu, -5.0/3) -
 			(15152.0/3031875*math.Pow(ar, 5)+1088.0/121275*ar*ar)*math.Pow(2, 1.0/3)*math.Pow(nu, -7.0/3)
-		c := cmplx.Sqrt(complex(r, 0))
-		xInit[i] = real(c)
+		if r < 0 {
+			r = 0
+		}
+		airyrts[i] = math.Sqrt(r)
```
Comment (vladimir-ch) [inline quad/hermite.go]: Call sqrt only for positive r?

**Comment by vladimir-ch** [comment]
LGTM

File: quad/hermite.go (hunk 1 above)
Comment (btracey) [inline quad/hermite.go]: I don't think so. The difficulty of using it in parallel go way up for not that much benefit.

File: quad/hermite.go (hunk 2 above)
Comment (btracey) [inline quad/hermite.go]: Will do in next PR.

**Pull request merged**